src/messaging/ExchangeContext.cpp

```diff
@@ -300,6 +300,27 @@ bool ExchangeContext::MatchExchange(SecureSessionHandle session, const PacketHea
         && (payloadHeader.IsInitiator() != IsInitiator());
 }
 
+void ExchangeContext::OnConnectionExpired()
+{
+    // Reset our mSecureSession to a default-initialized (hence not matching any
+    // connection state) value, because it's still referencing the now-expired
+    // connection.  This will mean that no more messages can be sent via this
+    // exchange, which seems fine given the semantics of connection expiration.
+    mSecureSession = SecureSessionHandle();
```

Comment:
After we reset the SessionHandle, this exchange can not be used to send any message.

Let's go back the the original issue.

After we send AddRootCert command and receive the response to AddRootCert, the connection is closed. 
Now we don't close the exchange, but reset the SessionHandle of the exchange.

On which exchange we send AddNetwork Command? My understand is we should continue to use the same exchange to send this command and close the exchange when we reach the end of conversation  (Received the response of AddNetwork Command)

Reply:
> On which exchange we send AddNetwork Command?

A different one.  Each new Invoke is a new exchange in the spec.